src/ORM/Table.php

```diff
@@ -1349,6 +1349,13 @@ public function save(EntityInterface $entity, $options = [])
             $success = $connection->transactional(function () use ($entity, $options) {
                 return $this->_processSave($entity, $options);
             });
+            if ($success) {
+                if (!$connection->inTransaction()) {
+                    $this->dispatchEvent('Model.afterSaveCommit', compact('entity', 'options'));
+                }
+                $entity->isNew(false);
+                $entity->source($this->registryAlias());
```

Comment:
Because they were earlier done in _processSave() after triggering Model.afterSave (removed lines 1422 and 1423 of diff). If they were not moved one would not be able to check whether it's a new entity in afterSaveCommit callback.